log/zap/zap_sugar_logger_test.go

```diff
@@ -0,0 +1,116 @@
+package zap_test
+
+import (
+	"encoding/json"
+	kitzap "github.com/go-kit/kit/log/zap"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"strings"
+	"testing"
+)
+
+type tbWriter struct {
+	tb testing.TB
+	sb strings.Builder
+}
+
+func (w *tbWriter) Write(b []byte) (n int, err error) {
+	w.tb.Logf(string(b))
+	w.sb.Write(b)
+	return len(b), nil
+}
+
+var encoderConfig = zap.NewDevelopmentEncoderConfig()
+
+func newLogger(writer zapcore.WriteSyncer, enabler zapcore.LevelEnabler) *zap.Logger {
+	return zap.New(
+		zapcore.NewCore(zapcore.NewJSONEncoder(encoderConfig), writer, enabler),
+		zap.Development())
+}
+
+type testCase struct {
+	level zapcore.Level
+	kvs   []interface{}
+	want  map[string]string
+}
+
+var testCases = []testCase{
+	{
+		zapcore.DebugLevel,
+		[]interface{}{"key1", "value1"},
+		map[string]string{
+			encoderConfig.LevelKey: "DEBUG",
+			"key1":                 "value1",
+		}},
+	{
+		zapcore.InfoLevel,
+		[]interface{}{"key2", "value2"},
+		map[string]string{
+			encoderConfig.LevelKey: "INFO",
+			"key2":                 "value2",
+		}},
+	{
+		zapcore.WarnLevel,
+		[]interface{}{"key3", "value3"},
+		map[string]string{
+			encoderConfig.LevelKey: "WARN",
+			"key3":                 "value3",
+		}},
+	{
+		zapcore.ErrorLevel,
+		[]interface{}{"key4", "value4"},
+		map[string]string{
+			encoderConfig.LevelKey: "ERROR",
+			"key4":                 "value4",
+		}},
+	{
+		zapcore.DPanicLevel,
+		[]interface{}{"key5", "value5"},
+		map[string]string{
+			encoderConfig.LevelKey: "DPANIC",
+			"key5":                 "value5",
+		}},
+	{
+		zapcore.PanicLevel,
+		[]interface{}{"key6", "value6"},
+		map[string]string{
+			encoderConfig.LevelKey: "PANIC",
+			"key6":                 "value6",
+		}},
+}
+
+func log(t *testing.T, logger *zap.Logger, testCase testCase) {
+	shouldPanic := testCase.level >= zapcore.DPanicLevel
+	if shouldPanic {
+		defer func() {
+			if x := recover(); x == nil {
+				t.Errorf("test level %v should panic", testCase.level)
+			}
+		}()
+	}
+	kitLogger := kitzap.NewZapSugarLogger(logger, testCase.level)
+	kitLogger.Log(testCase.kvs...)
+}
+
+func TestZapSugarLogger(t *testing.T) {
```

Comment:
Please put this test function at the top of the file so the top level code comes first and its dependencies are later. I believe it helps code readability to start with structure and follow with details rather than to start with details that are out of context.